packages/lodestar/src/network/reqResp.ts

```diff
@@ -161,11 +161,20 @@ export class ReqResp extends (EventEmitter as IReqEventEmitterClass) implements
     const {stream} = await this.libp2p.dialProtocol(peerInfo, protocol) as {stream: Stream};
     return await new Promise((resolve, reject) => {
       this.logger.verbose(`send ${method} request to ${peerInfo.id.toB58String()}`);
-      const responseTimer = setTimeout(() => reject(new Error(ERR_RESP_TIMEOUT)), RESP_TIMEOUT);
+      let responseTimer = setTimeout(() => reject(new Error(ERR_RESP_TIMEOUT)), TTFB_TIMEOUT);
       pipe(
         [this.encoder.encodeRequest(method, body)],
         stream,
         this.encoder.decodeResponse(method),
+        function handleTimeout(source: AsyncIterable<ResponseBody>): AsyncIterable<ResponseBody> {
```

Comment:
can you extract this as separate method instead.

Reply:
One thing to keep in mind (if you extract this function)
I think we will soon want to change the function signature of `sendRequest` to expose the async iterable (vs the arrayish promise).

Reply:
yeah we may want that when there's a need for `initial sync`, i.e. `BeaconBlockByRange` request. At that point we may want to have 2 different `sendRequest` versions, 1 returns an Array, other returns an Async Iterable.

From what I see, the current `sendRequest` interface works fine with `GoodBye`, `Status` and `BeaconBlockByRoot` (since we only request a single root/block atm).